src/backend/aspen/api/authz.py

```diff
@@ -163,3 +164,41 @@ async def get_authz_session(
     session: AsyncSession = Depends(get_db),
 ) -> AuthZSession:
     return AuthZSession(session, auth_context)
+
+
+async def get_oso(session: AsyncSession = Depends(get_db)) -> AsyncOso:
+    oso = AsyncOso()
+    oso.set_data_filtering_adapter(AsyncSqlAlchemyAdapter(session))
+    register_classes(oso)
+    await oso.load_files(
+        [Path.joinpath(Path(__file__).parent.absolute(), "policy.polar")]
+    )
+    return oso
+
+
+class AuthorizedSession:
+    def __init__(self, privilege: str, model: idbase):
+        self.privilege = privilege
+        self.model = model
+
+    async def __call__(
+        self,
+        auth_context: AuthContext = Depends(get_auth_context),
+        oso: AsyncOso = Depends(get_oso),
+    ):
+        self.auth_context = auth_context
+        self.oso = oso
+        return await self.authorized_query()
```

Comment:
This should be returning an authorized query from AuthZSession.authorized_query() instead of an oso query directly:
https://github.com/chanzuckerberg/czgenepi/blob/trunk/src/backend/aspen/api/authz.py#L138

Reply:
yeah, my bad